src/client/ui/EditorContainer.js

```
@@ -285,61 +289,73 @@ class EditorContainer extends Component {
+    this.showDialog(ProgressDialog, {
+      title: "Exporting Scene",
+      message: "Exporting scene..."
+    });
+
+    try {
+      // Export current editor scene using THREE.GLTFExporter
+      const { json, buffers, images } = await this.props.editor.exportScene();
+
+      // Ensure the output directory exists
+      await this.props.project.mkdir(outputPath);
+
+      // Write the .gltf file
+      const scene = this.props.editor.scene;
+      const gltfPath = outputPath + "/" + scene.name + ".gltf";
+      await this.props.project.writeJSON(gltfPath, json);
+
+      // Write .bin files
+      for (const [index, buffer] of buffers.entries()) {
+        if (buffer !== undefined) {
+          const bufferName = json.buffers[index].uri;
+          await this.props.project.writeBlob(outputPath + "/" + bufferName, buffer);
+        }
       }
-    }
 
-    // Write image files
-    for (const [index, image] of images.entries()) {
-      if (image !== undefined) {
-        const imageName = json.images[index].uri;
-        await this.props.project.writeBlob(outputPath + "/" + imageName, image);
+      // Write image files
+      for (const [index, image] of images.entries()) {
+        if (image !== undefined) {
+          const imageName = json.images[index].uri;
+          await this.props.project.writeBlob(outputPath + "/" + imageName, image);
+        }
       }
-    }
 
-    // Run optimizations on .gltf and overwrite any existing files
-    await this.props.project.optimizeScene(gltfPath, gltfPath);
+      // Run optimizations on .gltf and overwrite any existing files
+      await this.props.project.optimizeScene(gltfPath, gltfPath);
+    } catch (e) {
+      this.showDialog(ErrorDialog, {
+        title: "Error Exporting Scene",
+        message: e.message
+      });
+
+      console.error(e);
+    }
 
-    // Close modal
-    this.setState({ openModal: null });
+    this.hideDialog();
```

This hides the error dialog immediately in the error case.